src/frontend/src/scheduler/plan_fragmenter.rs

```diff
@@ -365,7 +381,18 @@ impl BatchPlanFragmenter {
                     builder.root = Some(Arc::new(execution_plan_node));
                 }
                 // Check out the comments for `has_table_scan` in `QueryStage`.
-                builder.has_table_scan |= node.node_type() == PlanNodeType::BatchSeqScan;
+                if let Some(scan_node) = node.as_batch_seq_scan() {
+                    let table_desc = scan_node.logical().table_desc();
+                    assert!(
+                        builder.table_scan_info.is_none(),
+                        "multiple table scan inside a stage"
+                    );
```

Comment:
It's possible that one stage has  more than one table scans, for example:
```
select * from t1, t2 where t1.a = t2.a
```
When t1 and t2 are both distributed by `a` and colocated in same node, we have following plan:
```
                          Exchange
                                 |
                           HashJoin
                   /                           \
              TableScan(t1)       TableScan(t2)
                
```

Reply:
Currently it's not possible, since HashJoin has Exchange on both its inputs. What about doing it later when needed?

Reply:
Hmm, the PR is too large, so I'm ok to put it later.